crates/block-producer/src/batch_builder/batch.rs

```diff
@@ -141,10 +139,11 @@ impl TransactionBatch {
                     // If an unauthenticated note was found in the store, transform it to an
                     // authenticated one (i.e. erase additional note details
                     // except the nullifier)
-                    found_unauthenticated_notes
-                        .get(&input_note_header.id())
-                        .map(|_path| InputNoteCommitment::from(input_note.nullifier()))
-                        .unwrap_or_else(|| input_note.clone())
+                    if found_unauthenticated_notes.contains_note(&input_note_header.id()) {
+                        InputNoteCommitment::from(input_note.nullifier())
+                    } else {
+                        input_note.clone()
+                    }
```

Comment:
What isn't clear to me is where/what the note inclusion proof should be used for?

Reply:
They are not currently used, but once we have batch/block kernels, they will be used in them for proving that these notes are actually authenticated.